lib/core/viewer/components/viewer.component.ts

```diff
@@ -660,42 +657,37 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
         if (!rendition) {
             renditionId = 'imgpreview';
             rendition = supported.list.entries.find(obj => obj.entry.id.toLowerCase() === renditionId);
-
-            if (!rendition) {
-                return null;
-            }
         }
 
-        const status = rendition.entry.status.toString();
+        if (rendition) {
+            const status = rendition.entry.status.toString();
 
-        if (status === 'CREATED') {
-            return rendition;
-        } else if (status === 'NOT_CREATED') {
-            try {
-                await this.apiService.renditionsApi.createRendition(nodeId, { id: renditionId });
-                return await this.waitRendition(nodeId, renditionId, 0);
-            } catch (err) {
-                this.logService.error(err);
-                return null;
-            }
+             if (status === 'NOT_CREATED') {
+                try {
+                    await this.apiService.renditionsApi.createRendition(nodeId, { id: renditionId });
+                    rendition =  await this.waitRendition(nodeId, renditionId, 0);
+                } catch (err) {
+                    this.logService.error(err);
+                }
+             }
         }
+
+        return rendition;
     }
 
     private async waitRendition(nodeId: string, renditionId: string, retries: number): Promise<RenditionEntry> {
         const rendition = await this.apiService.renditionsApi.getRendition(nodeId, renditionId);
 
-        if (retries > this.maxRetries) {
-            return null;
-        }
-
-        const status = rendition.entry.status.toString();
+        if (this.maxRetries < retries) {
```

Comment:
In case of `else` the promise will never finish?

Reply:
In case of `else`, the waitRendition async function will be an undefined Promise. Is not necessary to cover that case.